config/routes.rb

```diff
@@ -1,10 +1,15 @@
 Rails.application.routes.draw do
+  root to: 'users#index'
   get '/users', to: 'users#index', as: 'users'
   get '/users/:id', to: 'users#show', as: 'user'
   get '/users/:user_id/posts', to: 'posts#index', as: 'user_posts'
+  get '/users/:user_id/posts/new', to: 'posts#new', as: 'new_post'
+  post '/users/:user_id/posts', to: 'posts#create', as: 'create_post'
   get '/users/:user_id/posts/:id', to: 'posts#show', as: 'user_post'
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
-  # Defines the root path route ("/")
-  # root "articles#index"
+  resources :users, only: [] do
+  resources :posts, only: [] do
+    resources :comments, only: [:new, :create]
+    resources :likes, only: [:new, :create]
+  end
+end
 end
```

Comment:
[Optional] - This requirement is marked optional but highly recommended 💪

By simply adding `resources :posts`

This line will automatically generate the 8 routes. 
```
get    '/posts'          => 'posts#index'
get    '/posts/new'      => 'posts#new'
get    '/posts/:id'      => 'posts#show'
post   '/posts'          => 'posts#create'
get    '/posts/:id/edit' => 'posts#edit'
patch  '/posts/:id'      => 'posts#update'
put    '/posts/:id'      => 'posts#update'
delete '/posts/:id'      => 'posts#destroy'
```
[click here to learn more](https://launchschool.com/books/demystifying_rails/read/routes_and_resources#resourcenotation) 😊

So, I would suggest that we remove these lines from your code. line `3` to `8`.

```ruby
get '/users', to: 'users#index', as: 'users'
get '/users/:id', to: 'users#show', as: 'user'
get '/users/:user_id/posts', to: 'posts#index', as: 'user_posts'
get '/users/:user_id/posts/new', to: 'posts#new', as: 'new_post'
post '/users/:user_id/posts', to: 'posts#create', as: 'create_post'
get '/users/:user_id/posts/:id', to: 'posts#show', as: 'user_post'
```

Also, kindly remove the empty `[]` so that everything can work as expected 👍 

![image](https://github.com/DianaBeki/Blog-app/assets/78251927/0ecdc239-bfa7-467b-bb91-114a766405c1)

```ruby
resources :users do
    resources :posts do
      resources :comments, only: [:new, :create]
      resources :likes, only: [:new, :create]
    end
end
```